src/Tests/AuthoringTest/Program.cs

```diff
@@ -1865,6 +1865,30 @@ public sealed class CustomInterfaceGuidClass : ICustomInterfaceGuid
     {
         public string HelloWorld() => "Hello World!";
     }
+
+    public sealed class NonActivatableType
+    {
+        private readonly string _text;
+
+        // This should not be referenced by the generated activation factory
+        internal NonActivatableType(string text)
+        {
+            _text = text;
+        }
+
+        public string GetText()
+        {
+            return _text;
+        }
+    }
+
+    public static class NonActivatableFactory

```

Comment:
Can you add another test case or modify this one such that it is a sealed class with an internal constructor but also a static method that returns the object from the internal constructor.

Reply:
Done! 🙂